### PR TITLE
Cleanup axis3d.Axis.draw

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Remove a bunch of redundant lookups and computations.

Restructured code a bit.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by oscargus** [comment]
The test failure is caused by https://github.com/gristlabs/asttokens/issues/85

Not sure what the solution is:
1. Install `astroid`
2. Mark `asttokens!=2.0.6`
3. Wait for `asttokens` 2.0.7.

As we do not have any (direct) dependency on `asttokens` it doesn't really seem to make sense to select 2.

**Comment by alexmojaki** [comment]
asttokens 2.0.7 is released

**Comment by oscargus** [comment]
> asttokens 2.0.7 is released

Great, thanks!

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 1]
```diff
@@ -426,57 +419,61 @@ def draw(self, renderer):
...
-            self.gridlines.draw(renderer)
-
-        # Draw ticks:
-        tickdir = self._get_tickdir()
-        tickdelta = deltas[tickdir]
-        if highs[tickdir]:
-            ticksign = 1
-        else:
-            ticksign = -1
+        if len(ticks):
```
Comment (anntzer) [inline lib/mpl_toolkits/mplot3d/axis3d.py:422]: I prefer not including these kinds of checks (which lead to a large block being extra indented, and forcing the reader to keep track of when the condition stops to apply) unless it is either really necessary, or if the gain in time is really large when the block can be checked, which seems unlikely here?

Comment (oscargus) [reply lib/mpl_toolkits/mplot3d/axis3d.py]: OK! I reverted that.

File: lib/mpl_toolkits/mplot3d/axis3d.py [hunk 2]
```diff
@@ -435,47 +432,46 @@ def draw(self, renderer):
...
 
         # Draw ticks:
         tickdir = self._get_tickdir()
         tickdelta = deltas[tickdir]
-        if highs[tickdir]:
-            ticksign = 1
-        else:
-            ticksign = -1
-
+        tickdelta = tickdelta if highs[tickdir] else -tickdelta
```
Comment (QuLogic) [inline lib/mpl_toolkits/mplot3d/axis3d.py:445]: Could be joined?
```suggestion
        tickdelta = deltas[tickdir] if highs[tickdir] else -deltas[tickdir]
```

**Review by QuLogic — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]
After addressing @QuLogic ’s comment.

File: lib/mpl_toolkits/mplot3d/axis3d.py (hunk 2 above)
Comment (oscargus) [reply lib/mpl_toolkits/mplot3d/axis3d.py:445]: Fixed!

**Pull request merged**